sdk/src/nonce_keyed_account.rs

```diff
@@ -43,19 +47,33 @@ impl<'a> NonceKeyedAccount for KeyedAccount<'a> {
         &self,
         recent_blockhashes: &RecentBlockhashes,
         signers: &HashSet<Pubkey>,
+        invoke_context: &mut dyn InvokeContext,
     ) -> Result<(), InstructionError> {
         if recent_blockhashes.is_empty() {
+            ic_msg!(
+                invoke_context,
+                "Advance nonce account: recent blockhash list is empty",
+            );
             return Err(NonceError::NoRecentBlockhashes.into());
         }
 
         let state = AccountUtilsState::<Versions>::state(self)?.convert_to_current();
         match state {
             State::Initialized(data) => {
                 if !signers.contains(&data.authority) {
+                    ic_msg!(
+                        invoke_context,
+                        "Advance nonce account: Account {} must be a signer",
+                        data.authority
+                    );
                     return Err(InstructionError::MissingRequiredSignature);
                 }
                 let recent_blockhash = recent_blockhashes[0].blockhash;
                 if data.blockhash == recent_blockhash {
+                    ic_msg!(
+                        invoke_context,
+                        "Advance nonce account: nonce is not expired"
```

Comment:
```suggestion
                        "Advance nonce account: nonce can only advance once per slot"
```
Might be more clear?